src/utils/cleaning_data.py

```diff
@@ -0,0 +1,233 @@
+    elif upper_threshold is None:
+        filtered_df = df[df['missing_percentage'] < lower_threshold]
+    else:
+        filtered_df = df[
+            (df['missing_percentage'] > lower_threshold) & 
+            (df['missing_percentage'] < upper_threshold)
+        ]
+
+    # Return a list of the column names in the filtered DataFrame
+    return list(filtered_df.index)
+
+
+def fill_by_mode(data, columns_name) -> str:
+    """
+    Fill missing values in a pandas DataFrame column with the mode value.
+    
+    Args:
+    - data: A pandas DataFrame with the column to fill
+    - columns_name: The name of the column to fill
+    
+    Returns:
+    - A string indicating the success of the operation.
+    """
+    
+    # Fill missing values in the specified column using the mode
+    mode_value = data[columns_name].mode().iloc[0,0]   # Compute the mode value
+    data[columns_name] = data[columns_name].fillna(mode_value)   # Fill missing values with mode value
+    
+    # Return a success message
+    return "Provided column successfully filled by mode value."
+
+
+def filter_indirect_missing_columns(df, columns, lower_threshold=None, upper_threshold=None) -> list:
+    """
+    Filter a pandas DataFrame based on the percentage of indirect missing values for a list of columns.
+
+    Args:
+    - df: A pandas DataFrame
+    - columns: A list of column names to filter
+    - lower_threshold: A float indicating the lower threshold for the percentage of indirect missing values
+    - upper_threshold: A float indicating the upper threshold for the percentage of indirect missing values
+
+    Returns:
+    - A list of column names in the filtered DataFrame
+    """
+    
+    # Convert selected columns to numeric data type
+    df[columns] = df[columns].apply(pd.to_numeric, errors='coerce')
+
+    # Filter the DataFrame based on the user-provided thresholds for each column
+    if lower_threshold is None:
+        filtered_df = df.loc[df[columns].max(axis=1) > upper_threshold]
+
+    elif upper_threshold is None:
+        filtered_df = df.loc[df[columns].min(axis=1) < lower_threshold]
+    
+    else:
+        filtered_df = df.loc[
+            (df[columns].max(axis=1) > lower_threshold) & 
+            (df[columns].min(axis=1) < upper_threshold)
+        ]
+
+    # Return a list of the column names in the filtered DataFrame
+    return list(filtered_df.index)
+
+
+def _is_one_hot_encoded(df, col_name):
+    """
+    Returns True if the given column is one-hot encoded in the DataFrame, and False otherwise.
+
+    Args:
+    - df: A pandas DataFrame
+    - col_name: A string representing the name of the column
+
+    Returns:
+    - True if the column is one-hot encoded, and False otherwise
+    """
+    unique_vals = df[col_name].unique()
+
+    # Check if the column has only two unique values
+    if len(unique_vals) == 3:
+        return True
+
+    # Check if the column name contains an underscore followed by a unique value
```

Comment:
it's better to make this in a separate function; to reduce code redundancy. 
(as it may be needed in other steps (such as EDA))

Reply:
I add _is_one_hot_encoded_ function to check whether the given columns are encoded features or not.